cpp/models/ode_secir/model.h

```diff
@@ -328,6 +329,264 @@ double get_infections_relative(const Simulation<Base>& sim, double /*t*/, const
+    }
+
+    double season_val                        = (1 + params.template get<Seasonality>() *
+                                 sin(pi * (std::fmod((sim.get_model().parameters.template get<StartDay>() +
+                                                      sim.get_result().get_time(t_idx)),
+                                                     365.0) /
+                                               182.5 +
+                                           0.5)));
+    ContactMatrixGroup const& contact_matrix = sim.get_model().parameters.template get<ContactPatterns>();
+
+    Eigen::MatrixXd cont_freq_eff(num_groups, num_groups);
+    Eigen::MatrixXd riskFromInfectedSymptomatic_derivatives(num_groups, num_groups);
+    Eigen::VectorXd divN(num_groups);
+    Eigen::VectorXd riskFromInfectedSymptomatic(num_groups);
+    Eigen::VectorXd rateINS(num_groups);
+
+    for (mio::AgeGroup k = 0; k < (mio::AgeGroup)num_groups; k++) {
+        double temp = sim.get_result().get_value(
+                          t_idx)[sim.get_model().populations.get_flat_index({k, InfectionState::Susceptible})] +
+                      sim.get_result().get_value(
+                          t_idx)[sim.get_model().populations.get_flat_index({k, InfectionState::Exposed})] +
+                      sim.get_result().get_value(
+                          t_idx)[sim.get_model().populations.get_flat_index({k, InfectionState::InfectedNoSymptoms})] +
+                      sim.get_result().get_value(
+                          t_idx)[sim.get_model().populations.get_flat_index({k, InfectionState::InfectedSymptoms})] +
+                      sim.get_result().get_value(
+                          t_idx)[sim.get_model().populations.get_flat_index({k, InfectionState::InfectedSevere})] +
+                      sim.get_result().get_value(
+                          t_idx)[sim.get_model().populations.get_flat_index({k, InfectionState::InfectedCritical})] +
+                      sim.get_result().get_value(
+                          t_idx)[sim.get_model().populations.get_flat_index({k, InfectionState::Recovered})];
+        if (temp == 0) {
+            temp = 1;
+        }
+        divN[(size_t)k] = 1 / temp;
+
+        riskFromInfectedSymptomatic[(size_t)k] = smoother_cosine(
+            test_and_trace_required, params.template get<TestAndTraceCapacity>(),
+            (params.template get<TestAndTraceCapacity>()) * 5, params.template get<RiskOfInfectionFromSymptomatic>()[k],
+            params.template get<MaxRiskOfInfectionFromSymptomatic>()[k]);
+
+        rateINS[(size_t)k] =
+            0.5 / (params.template get<IncubationTime>()[k] - params.template get<SerialInterval>()[(mio::AgeGroup)k]);
+
+        for (mio::AgeGroup l = 0; l < (mio::AgeGroup)num_groups; l++) {
+            if (test_and_trace_required < params.template get<TestAndTraceCapacity>() ||
+                test_and_trace_required > 5 * params.template get<TestAndTraceCapacity>()) {
```

Comment:
For all models, we should add parameters that describe the multiplicative factor used for the TestAndTraceCapacity as we use different realizations in secir and secirvvs simulations. This is inconsistent but not a problem from this new feature.